instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt

```diff
@@ -372,6 +385,59 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
         countDownLatch.await()
     }
 
+    /**
+     * This test ensures that we're not crashing when parsing alternative routes metadata
+     * in a specific case (caught by crashlytics) that involved
+     * Nav Native reporting the [AlternativeRouteIntersection.geometryIndexInRoute] of
+     * [AlternativeRouteMetadata.forkIntersectionOfAlternativeRoute] to be out of bounds
+     * for the shape points collection of that particular route.
+     */
+    @Test
+    fun regression_test_NAVAND_692() = sdkTest {
+        val primaryRoute = createRoute(
+            responseJson = R.raw.route_response_japan_1,
+            requestUrlJson = R.raw.route_response_japan_1_url,
+        ).first()
+        val alternativeRoute = createRoute(
+            responseJson = R.raw.route_response_japan_2,
+            requestUrlJson = R.raw.route_response_japan_2_url,
+        ).first()
+        val mapboxNavigation = MapboxNavigation(
+            NavigationOptions.Builder(activity.applicationContext)
+                .accessToken(getMapboxAccessTokenFromResources(activity))
+                .build()
+        )
+
+        mapboxNavigation.setNavigationRoutes(
+            listOf(
+                primaryRoute,
+                alternativeRoute
+            )
+        )
+        val routesUpdate = mapboxNavigation.routesUpdates()
+            .take(1)
+            .map { it.navigationRoutes }
+            .toList().first()
+
+        val options = MapboxRouteLineOptions.Builder(activity)
+            .build()
+        val routeLineApi = MapboxRouteLineApi(options)
+        val result = routeLineApi.setNavigationRoutes(
+            newRoutes = routesUpdate,
+            alternativeRoutesMetadata = mapboxNavigation.getAlternativeMetadataFor(routesUpdate)
+        )
+
+        // we're expecting to ignore the attempt to hide the alternative until the deviation point,
+        // hence the cleared offset
+        assertEquals(
+            0.0,
+            result.value!!.alternativeRouteLinesData[0].dynamicData.trimOffset!!.offset,
+            0.0000000001
+        )
+
+        mapboxNavigation.onDestroy()
```

Comment:
It won't be invoked if the test fails. It can be done in `@After` or by using `MapboxNavigationRule`.

Reply:
In the end, this hardly matters since the test would have to be fixed first anyway, doesn't it?

I'm thinking because sometimes unit tests are very localized, and I want to avoid mocking, for example, statics or `object`s for the whole test class when I just need them mocked in a single test.

I guess this is more of the problem of the implementation code organization and structure though.

Reply:
> In the end, this hardly matters since the test would have to be fixed first anyway, doesn't it?

Won't it affect other tests? You'll try to create a new `mapboxNavigation` instance when the prev one was not destroyed (because `hasInstance` is static).

Reply:
Possibly, but since there's a failing (crashing) test, it has to be adjusted anyway and the test suite has to be re-run. This fails to follow the principle of keeping tests in full isolation but that's a general problem of static entities.

Reply:
But there is a difference: you see one failing test or 10 failing tests, while 9 of them failed only because the first one failed. It's possible you'll waste time trying to figure out why those 9 failed.
I don't see a problem with using `@After`. We use `@Before` a lot.

Reply:
On the other hand we have 9 tests that run with a mocked static while they'd expect to work with a non-modified instance. This example is not applicable to the case of `MapboxNavigation`, I'm referring to a general mocking of statics and objects.

I'm just thinking out loud about potential edge cases, In general I agree with `@Before`/@After` usage as even for statics mocking has no impact unless specific function is explicitly mocked, but the potential for misuse is there.

Reply:
We can create `MockedStaticRule` so that we don't have to write `@Before` and `@After` every time just for static mocking.